Add Lando documentation to Learn

fembuelita opened the pull request:
I've added a rough draft of the Lando configuration for UF. I'm submitting this to `develop` instead of `hotfix` as I would love someone who doesn't already have lando installed to try this out and verify it works.

I will try running this on a windows machine as well but it will take me some time to procure that. I've added notes regarding limitations of Windows machines here until then.

fembuelita committed:
Add rough draft of Lando doc

fembuelita committed:
Disable custom php ini settings

fembuelita committed:
Bug fix to lando composer install command

fembuelita committed:
Add npm build steps to lando config

fembuelita committed:
Comment out apache rewrite support

fembuelita committed:
Remove restart call for apache since its not needed and clarify instructions for lemp

fembuelita committed:
clarify windows support for lando

fembuelita commented:
This has been fully tested on a new Ubuntu install and Mac install. There are reports that Lando doesn't play well with Windows and official docs from Lando/Docker about strict requirements for Windows. I've updated the notice at the top of page to reflect strict requirements for Windows.

fembuelita commented:
Is there something I can do to help merge this into the documentation? Lando is arguably a better solution than Docker since it simplifies the process so much and is built specifically for PHP apps that run on Docker.

lcharette commented:
I never gave Lando a try. Did someone else testes this new doc ?

Silic0nS0ldier commented:
Lando is used extensively by the PHP teams (myself included) at my work. I had actually seen this PR before but didn't give any input as it was still being evaluated at the time (and very much a word in progress).

Haven't run the sample configuration here locally yet.

I had a quick look over the PR, I don't think git should be included in the exposed tooling. I can see a number of potential problems arising from encouraging this (unexpected behaviour due to host configuration not being picked up, lockfile weirdness, extreme performance hit on MacOS and Windows [recommended WSL2 setups would be fine] due to slow volume mounts, increased risk of change propagation errors).

fembuelita committed:
Remove git tooling exposure; bump node -> 12 LTS

fembuelita commented:
@Silic0nS0ldier that's fair pushback. I removed the git tooling exposure and I also bumped Node 11 -> Node 12.18 (LTS)

Silic0nS0ldier commented:
Content LGTM, haven't tested however. Once I've the time, should be a quick check. Its all probably good.

Silic0nS0ldier commented on pages/03.installation/01.environment/04.lando/docs.md:
````diff
@@ -0,0 +1,188 @@
+
+When the application boots successfully, you'll see something like: 
+
+```
````
This is wrong. I'd guess a leftover from a previous draft or older lando version maybe. It should be something like;
```
   ___                      __        __        __     __        ______
  / _ )___  ___  __ _  ___ / /  ___ _/ /_____ _/ /__ _/ /_____ _/ / / /
 / _  / _ \/ _ \/  ' \(_-</ _ \/ _ `/  '_/ _ `/ / _ `/  '_/ _ `/_/_/_/ 
/____/\___/\___/_/_/_/___/_//_/\_,_/_/\_\\_,_/_/\_,_/_/\_\\_,_(_|_|_)  
                                                                       

Your app has started up correctly.
Here are some vitals:

 NAME            userfrosting                       
 LOCATION        /your/app/location
 SERVICES        appserver, database, pma           
 APPSERVER URLS  https://localhost:32773            
                 http://localhost:32774             
                 http://userfrosting.lndo.site/     
                 https://userfrosting.lndo.site/    
 PMA URLS        http://localhost:32776             
                 http://pma.userfrosting.lndo.site/ 
```

Silic0nS0ldier commented on pages/03.installation/01.environment/04.lando/docs.md:
````diff
@@ -0,0 +1,188 @@
+
+#### Sample Lando Configuration File
+
+```yml
````
It would be useful to include the `bakery` CLI in the Lando tooling.
```
  bakery:
    service: appserver
    cmd: php bakery
```

Silic0nS0ldier commented on pages/03.installation/01.environment/04.lando/docs.md:
````diff
@@ -0,0 +1,188 @@
+
+#### Sample Lando Configuration File
+
+```yml
````
Since we are already including helper services (PHP My Admin) we might as well go all the way. Lando supports the MailHog service (https://docs.lando.dev/config/mailhog.html) which would be a fantastic addition.

Silic0nS0ldier replied:
Seems some work will need to be done on UserFrosting first to make using the MailHog service easier to integrate with.

Silic0nS0ldier reviewed (changes requested):
This works and overall LGTM. There are some things I would love to see but I won't let them block merging. 

The only thing I'd want to see changed before merging is updated the message in `1st Time Lando Start-up`.

Silic0nS0ldier commented on pages/03.installation/01.environment/04.lando/docs.md:
```diff
@@ -0,0 +1,188 @@
+  phpunit:
+    service: appserver
+    description: "Run PHP Unit tests: lando phpunit"
+  composer:
```
Based on https://docs.lando.dev/config/lamp.html#tooling, this should be included by default in the LAMP recipe.

Silic0nS0ldier committed:
Lando documentation updates
- Updated to reflect changes in stable Lando v3
- Removed documentation that effectively duplicated Lando's own
- More linking to supplemental external documentation
- Removed redundant steps, consolidated others
- Misc. fixes

Silic0nS0ldier committed:
Removed redundant syntax

Silic0nS0ldier commented:
Alright, I've made some changes based on my previous comments (and more as a result of having had a much closer look and actually road testing this properly).

I had a go at adding in the MailHog service however the current way UserFrosting is configured would mean code changes (config) would be required. Entirely possible, but work I'm too lazy/bored to do myself at this moment.

@lcharette I have no (some, technically) idea how this is handled post-merge so I'll leave the rest of this to you (or @alexweissman if around and not as clueless as I).

lcharette commented:
Merged and deployed on the current doc version !
